google-cloud-pubsub/lib/google/cloud/pubsub/topic/async_publisher.rb

```diff
@@ -147,10 +147,12 @@ def publish_batch_async topic_name, batch
             Concurrent::Future.new(executor: @thread_pool) do
               begin
                 grpc = @service.publish topic_name, batch.messages
+                publish_timestamp = Convert.time_to_timestamp Time.now
                 batch.items.zip(Array(grpc.message_ids)) do |item, id|
                   next unless item.callback
 
                   item.msg.message_id = id
+                  item.msg.publish_time = publish_timestamp
```

Comment:
Might be better to use `published_at` instead of the alias `publish_time`, to make it less likely that some maintainer misses this when searching for usages. Or reverse which is the impl and which is the alias.